### DISCUSSION TITLE
Best release to choose for a deployment

### DISCUSSION THREAD

**Discussion by nico3133** [actions/virtual-environments#9566] [General questions] [answered]
Hi all,

I've been successfully using Ubuntu and Windows images as build agents for Azure DevOps for a few months now. I just have one question: What's the best code repository to choose if I want to deploy updated images on my Azure?

On the homepage, I see a "Release" menu, which I assume contains the latest stable releases of the images as well as pre-releases. However, I also noticed the main branch on the homepage. Does the main branch contain the same files as the release versions? but with some improvement ?

I'm not sure if I was clear in my request, but if you need more information to help me, please don't hesitate to ask.

Thanks for all!

**Comment by mikhailkoliada** [comment, marked as answer]
Hey! As stated in [our docs](https://github.com/actions/runner-images?tab=readme-ov-file#faqs) we do not support building from the releases as we do not back port patches there, the best thing to do is to always use the main branch.

**Reply by nico3133** [reply]
Hello !
OK thanks for the answer, help me a lot !
